### ISSUE TITLE
"Gamerules lookup failed" error

### ISSUE THREAD

**Issue by Jugheadbomb** [Mikusch/friendlyfire#3]
Happens only with `SDKHookCB_Object_SpawnPost` callback, possibly related to map buildings (in that case Gamerules pointer might be invalid)

```cpp
L 03/16/2023 - 22:21:55: [SM] Exception reported: Gamerules lookup failed.
L 03/16/2023 - 22:21:55: [SM] Blaming: friendlyfire.smx
L 03/16/2023 - 22:21:55: [SM] Call stack trace:
L 03/16/2023 - 22:21:55: [SM]   [0] GameRules_GetProp
L 03/16/2023 - 22:21:55: [SM]   [1] Line 325, friendlyfire\sdkhooks.sp::SDKHookCB_Object_SpawnPost
L 03/16/2023 - 22:42:37: Error log file session closed.
```

**Comment by Mikusch** [comment]
Looks like a stupid Source quirk where map entities are spawned before gamerules ptr is initialized. I don't think this happens outside of the SpawnPost of objects, but I've put out a more thorough fix in #4.

Are you able to quickly test and verify that it works?

**Comment by Mikusch** [comment]
The linked PR actually turned out a bit bigger than I anticipated, however it should fix everything related to entity hooking

**Comment by Jugheadbomb** [comment]
It seems issue is not completed, error still occurs on last version.
```cpp
L 06/24/2023 - 16:03:37: [SM] Exception reported: Gamerules lookup failed.
L 06/24/2023 - 16:03:37: [SM] Blaming: friendlyfire.smx
L 06/24/2023 - 16:03:37: [SM] Call stack trace:
L 06/24/2023 - 16:03:37: [SM]   [0] GameRules_GetProp
L 06/24/2023 - 16:03:37: [SM]   [1] Line 37, friendlyfire\util.sp::IsFriendlyFireEnabled
L 06/24/2023 - 16:03:37: [SM]   [2] Line 329, friendlyfire\sdkhooks.sp::SDKHookCB_Object_SpawnPost
```

**Comment by Mikusch** [comment]
Probably a regression from one update I did, I'll look into it

**Comment by Mikusch** [comment]
Fixed with dc64eee2dde0a3a549034318498da42e14c70dd1
